src/core/eme/index.ts

```diff
@@ -135,53 +132,92 @@ function handleEncryptedEvent(
  * The EME handler can be given one or multiple systems and will choose the
  * appropriate one supported by the user's browser.
  * @param {HTMLMediaElement} video
- * @param {Object} keySystems
+ * @param {Array.<Object>} keySystems
  * @param {Subject} errorStream
  * @returns {Observable}
+ * TODO we should not return raw events, but document each one which can be
+ * returned.
  */
 function createEME(
   video : HTMLMediaElement,
   keySystems: IKeySystemOption[],
   errorStream: ErrorStream
-) : Observable<IMockMediaKeys|MediaKeys|ISessionEvent|Event> {
+) : Observable<ISessionEvent|Event> {
   if (__DEV__) {
     keySystems.forEach((ks) => assert.iface(ks, "keySystem", {
       getLicense: "function",
       type: "string",
     }));
   }
 
-  // get the MediaKeys element the associated options
-  // This is done as the "encrypted" event is not yet received for performance
-  const mediaKeysInfos$ = findCompatibleKeySystem(keySystems, instanceInfos)
-    .mergeMap((keySystemAccessInfos) => {
-      return createMediaKeysObs(keySystemAccessInfos.keySystemAccess)
-        .mergeMap(function prepareMediaKeysConfiguration(mediaKeys) {
-          const { keySystem } = keySystemAccessInfos;
-          const { serverCertificate } = keySystem;
-          setSessionStorage(keySystem);
+  /**
+   * Encrypted events may be distinct until initData changes.
+   * It is usefull to trigger session creation on two consecutive init datas.
+   *
+   * @type {Observable}
+   */
+  const encrypted$ = onEncrypted$(video)
+    .distinctUntilChanged(function isSameEncryptedEvent(prevEvt, evt) {
```

Comment:
Wouldn't there be a risk in possible future optimizations if encrypted events for the same initData happens but after a really long time between the two events ?

Like:
  1. playing encrypted event
  2. a "encrypted" event pops up
  3. create media keys, session an whatnot
  4. after a long time (and perhaps some unencrypted content between), another encrypted event with the same initData pops-up
  5. do nothing as it doesn't pass through this filter
  6. the browser can't unencrypt the content

The risk is pretty low but this logic still not seems very _logical_ when I think about it.

Wouldn't be the right logic to say that for a given initData:
  - No MediaKeySession are created
  - No MediaKeySession is pending to be created